pilot/pkg/xds/filters/filters.go

```diff
@@ -213,76 +199,6 @@ var (
 		},
 	}
 
-	SidecarInboundMetadataFilter = &hcm.HttpFilter{
```

Comment:
The downside of removing all of these is that we're now allocating a bunch of objects in a hot path of the control plane (before, they were static pointers). We should try and create a function that modifies the typed config if there are custom labels (this is the least common path)

Reply:
Technically I can prebuilt these filters when server initially started(that will bring more changes) and we need to assume that env flags won't change(this's fine in k8s/docker, not sure if someone run pilot in VM).

Reply:
env var changes are not supported for any env in Istio and don't really make sense (there is no way to change a processes env except from the process itself and Istiod never changes it)

Reply:
let me prebuilt it.

Reply:
updated.